modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java

```diff
@@ -124,56 +123,63 @@ public void onFailure(Exception ex) {
     @Override
     public void onCheckpoint(ScrollableHitSource.Checkpoint checkpoint, BulkByScrollTask.Status status) {
         assert checkpointThrottler != null;
-        checkpointThrottler.accept(Tuple.tuple(checkpoint, status));
+        assert finishedListener != null;
+
+        // TODO: Test
+        if (isDone.get() == false) {
```

Comment:
nit: This isDone check unnecessary, is already guaranteed by the ThrottlingConsumer.